lib/ts/recipe/session/recipeImplementation.ts

```diff
@@ -378,13 +410,74 @@ export default function getRecipeInterface(querier: Querier, config: TypeNormali
             return SessionFunctions.updateAccessTokenPayload(helpers, sessionHandle, newAccessTokenPayload);
         },
 
+        mergeIntoAccessTokenPayload: async function ({
+            sessionHandle,
+            accessTokenPayloadUpdate,
+        }: {
+            sessionHandle: string;
+            accessTokenPayloadUpdate: JSONObject;
+        }) {
+            const sessionInfo = await SessionFunctions.getSessionInformation(helpers, sessionHandle);
+            const updatedPayload = { ...sessionInfo.accessTokenPayload, ...accessTokenPayloadUpdate };
+            for (const key of Object.keys(accessTokenPayloadUpdate)) {
+                if (accessTokenPayloadUpdate[key] === null) {
+                    delete updatedPayload[key];
+                }
+            }
+            return this.updateAccessTokenPayload(sessionHandle, updatedPayload);
+        },
+
         getAccessTokenLifeTimeMS: async function (): Promise<number> {
             return (await getHandshakeInfo()).accessTokenValidity;
         },
 
         getRefreshTokenLifeTimeMS: async function (): Promise<number> {
             return (await getHandshakeInfo()).refreshTokenValidity;
         },
+
+        applyClaimBuilder: function <T>(input: {
+            sessionHandle: string;
+            claimBuilder: SessionClaimBuilder<T>;
+            userContext?: any;
+        }) {
+            const accessTokenPayloadUpdate = input.claimBuilder.applyToPayload(this.getUserId(), {}, input.userContext);
+            return this.mergeIntoAccessTokenPayload({
+                sessionHandle: input.sessionHandle,
+                accessTokenPayloadUpdate,
+                userContext: input.userContext,
+            });
+        },
+
+        setClaimValue: function <T>(input: {
```

Comment:
I think the name is different enough to avoid mistaking one for another + the different signatures help I think, but maybe the "apply" terminology isn't clear enough.
Some other name options (but all seem weirdly verbose):
1. `fetchAndAddClaimValue`
2. `fetchAndSetClaimValue`
3. `buildAndAddClaim`

Reply:
So applyClaim is only meant to be called when setting the claim in a session for the first time? If yes, how about calling it `initialiseAndSetClaim`?

Reply:
No, you can use it to "refresh" the claim from the DB as well.